File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/common/FaciaGalleryItem.java
package com.gu.fronts.integration.test.page.common;

import static com.gu.fronts.integration.test.fw.selenium.ByTestAttributeSelector.byTestAttribute;
import static com.gu.fronts.integration.test.page.util.PageElementHelper.findElementBy;
import static org.openqa.selenium.By.cssSelector;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class FaciaGalleryItem extends AbstractParentPage {
    public static final String GALLERY_ITEM_CONTAINER_ID = "gallery-item-container";
    private static final String GALLERY_PICTURE_ID = "gallery-picture";

    private WebElement rootElement;

    public FaciaGalleryItem(WebDriver webDriver, WebElement containerTopElement) {
        super(webDriver);
        this.rootElement = containerTopElement;
    }

    @Override
    public FaciaGalleryItem isDisplayed() {
        assertExistsAndDisplayed(rootElement);
        return this;
    }

    public GalleryOverlay clickPicture() {
        pictureElement().click();
        return pageFactory.initPage(webDriver, GalleryOverlay.class);
    }

    private WebElement pictureElement() {
        return findElementBy(rootElement, cssSelector(byTestAttribute(GALLERY_PICTURE_ID)));
    }

}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/FaciaGalleryItem.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.By.cssSelector
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

import com.gu.fronts.integration.test.fw.selenium.ByTestAttributeSelector.byTestAttribute
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed
import com.gu.fronts.integration.test.page.util.PageElementHelper.findElementBy
import com.gu.fronts.integration.test.page.common.FaciaGalleryItem.GALLERY_PICTURE_ID

object FaciaGalleryItem {

  val GALLERY_ITEM_CONTAINER_ID = "gallery-item-container"

  private val GALLERY_PICTURE_ID = "gallery-picture"
}

class FaciaGalleryItem(webDriver: WebDriver, containerTopElement: WebElement) extends FrontsParentPage(webDriver) {

  private var rootElement: WebElement = containerTopElement

  override def isDisplayed(): Boolean = {
    existsAndDisplayed(rootElement)
  }

  def clickPicture(): GalleryOverlay = {
    pictureElement().click()
    pageFactory.initPage(webDriver, classOf[GalleryOverlay])
  }

  private def pictureElement(): WebElement = {
    findElementBy(rootElement, cssSelector(byTestAttribute(GALLERY_PICTURE_ID)))
  }
}